UserApi/UserApi/Helper/UserProfileHelper.cs

```diff
@@ -34,8 +40,9 @@ public async Task<UserProfile> GetUserProfile(string filter)
         {
             var user = await _userAccountService.GetUserByFilter(filter);
 
-            if (user == null)
+            if (user == null) {
                 return null;
+            }
 
             var userGroupDetails = await _userAccountService.GetGroupsForUser(user.Id);
             var userGroups = GetUserGroups(userGroupDetails).ToList();
```

Comment:
Remove ToList() and change the called method param from List to IEnumerable

Reply:
Since I enumerate that collection several times I need to make sure it's been evaluated before passing it around so that's why I've got the ToList there. Same thing with the input parameter. I could have that as an enumerable but I'll need to re-map it anyway so I might as well pass it in as a collection.